cli/tests/test_comments.rs

```diff
@@ -246,12 +256,30 @@ fn test_delete_comments_in_range() {
     ]);
 
     // Get all comments and check there are none left
-    let after_deleting_all = cli.run([
-        "get",
-        "comments",
-        "--dataset",
-        dataset1.identifier(),
-        source.identifier(),
-    ]);
+    let after_deleting_all = get_comments_with_delay(
+        cli,
+        [
+            "get",
+            "comments",
+            "--dataset",
+            dataset1.identifier(),
+            source.identifier(),
+        ],
+        0,
+    );
     assert_eq!(after_deleting_all.lines().count(), 0);
 }
+
+fn get_comments_with_delay(cli: &TestCli, command: [&str; 5], expected_count: usize) -> String {
+    let mut result = cli.run(command);
+    for _ in 0..60 {
+        if result.lines().count() == expected_count {
+            break;
+        } else {
+            thread::sleep(Duration::from_secs(1));
+            result = cli.run(command);
+        }
+    }
+
+    result
+}
```

Comment:
Yeah agreed with Cristi this is a totally fine approach, we have something similar in some of our python tests here: https://github.com/reinfer/platform/blob/38ff0f6c5e964c41edfa19b84c98fd6261ad417d/exchange_online_sync/reinfer/exchange_online_sync/tests/test_sync.py#L353

You might also prefer to just pull in a crate with an already nice api + exponential backoff like `backoff`: https://lib.rs/crates/backoff

also nit: use `&[&str]` rather than `[&str; 5]` which is a very strict input requirement (fixed length)